core/src/main/java/org/frankframework/pipes/EscapePipe.java

```diff
@@ -72,26 +72,31 @@ public PipeRunResult doPipe(Message message, PipeLineSession session) throws Pip
 			throw new PipeRunException(this, "cannot open stream", e);
 		}
 
-		String substring = null;
-		String result = input;
+		String result = handleSubstrings(input);
+		return new PipeRunResult(getSuccessForward(), result);
+	}
+
+	private String handleSubstrings(String input) {
+		if (substringStart == null || substringEnd == null) {
+			return handle(input);
+		}
+
 		int i = -1;
-		int j = -1;
-		log.debug("substringStart [{}] substringEnd [{}] input [{}]", substringStart, substringEnd, input);
-		if (substringStart != null && substringEnd != null) {
-			i = input.indexOf(substringStart);
-			if (i != -1) {
-				j = input.indexOf(substringEnd, i);
-				if (j != -1) {
-					substring = input.substring(i + substringStart.length(), j);
-					substring = handle(substring);
-					result = input.substring(0, i + substringStart.length()) + substring + input.substring(j);
-				}
+		while ((i = input.indexOf(substringStart, i + 1)) != -1) {
```

Comment:
With this new solution, the while-loop is cycling through every character of the input string, even though perhaps there are no hits at all. That is not needed right? 

Perhaps an approach could work that works like this: search for first occurrence -> handle it. Check of second occurrence -> handle it (or if not found, stop). Etc. without looping over every character of the String.

Reply:
This loop is indeed rather sub-optimal.
1. You replace the string while working on it. Potentially multiple times.
2. You continue searching from the last found position of the start-string, instead of last found position of the end-string. (Although by changing the string in-place, that position is now invalid).

Much better to collect new string in a StringBuilder, and also the method `handle(content)` can append to that buffer.